### PR TITLE
Fixed handling of "@"-sign in file URLs.

### PR DESCRIPTION
**Pull request by svenpanne**
TL;DR: pipenv is broken if you try to install a package from a local file
and the path to it has an "@"-sign somewhere in it. (Jenkins workspaces!)

Long version: Consider the following scenario: Your `Pipfile` contains a
package description like

     coolpkg = {path = "./foo/bar/coolpkg.tar.gz"}

and the absolute path to the `Pipfile` is `/blah/blubb@2/baz/Pipfile`,
pipenv constructs the URI

     file:///blah/blubb@2/baz/foo/bar/coolpkg.tar.gz#egg=coolpkg

Things are OK so far, but then `Line.line_is_installable` uses
`split_ref_from_uri`, passing the resulting line to `is_installable_file`.
The problem is that `split_ref_from_uri` is too naive and thinks the part
after the "@" is a ref, passing only `file:///blah/blubb` to
`is_installable_file`, which of course fails.

This fix changes `split_ref_from_uri` so that it never considers an "@" in a
`file:` URL to be a ref, files don't have branches/tags/hashes, after all.
I am not sure if this is the nicest place to fix this, but *some* kind of
fix is urgently needed: [Jenkins](https://jenkins.io) uses "@2", "@3", ...
suffixes in its workspace names, which effectively makes pipenv unusable
when you have local packages and one of the most common CI tools. :-(

### CODE REVIEW

**Comment by techalchemy** [comment]
Would you mind adding a news entry in `news/168.bugfix.rst` with a short sentence describing the fix? Otherwise it seems correct, thanks!

**Comment by svenpanne** [comment]
Done...

**Comment by techalchemy** [comment]
perfect, thanks!

**Comment by techalchemy** [comment]
Interesting build failure.  You can ignore it for sure, it is either transient or irrelevant as it pertains to part of the code that isn't even in use. It's on the list to deprecate and delete that

**Comment by svenpanne** [comment]
The even more interesting part is that there were no failures before I rebased the 2nd version of the PR... 😉

**Comment by techalchemy** [comment]
And master is passing too, it makes no sense

I'll just trigger a rebuild

**Comment by techalchemy** [comment]
I'm going with pypi cache issue on azure?
